docs/modules/ROOT/pages/explanation/decisions/portal-api.adoc

```diff
@@ -0,0 +1,103 @@
+. Eventually, the foreign ERP system has generated the invoice.
+. Clients requesting a list of invoices will transparently query the ERP adapter which will then return a set of relevant invoices.
+====
+
+=== Kubernetes API
+
+Use the Kubernetes API server to leverage all the core-plumbing which already exists there: Authentication, RBAC, scalability, well-defined resource-model, extensibility with `CRDs`, aggregated API server and much more.
+This seems to be a common topic in the Cloud Native world, see for example the experiment called https://github.com/kcp-dev/kcp[kcp - a minimal Kubernetes API server].
+
+A CLI client could be there "for free" if wanted.
+The Kubernetes CLI client `kubectl` can display and work with the custom API endpoints of the Portal API out-of-the-box.
+
+Implementation-wise the core https://kubernetes.io/docs/concepts/extend-kubernetes/[extensibility features] of the Kubernetes API server are leveraged.
+This could mean that the API specification is written as CRDs and the backend functionality implemented as a https://kubernetes.io/docs/concepts/architecture/controller/[Kubernetes controller] or if the use-case can't be reached this way by using a custom https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/apiserver-aggregation/[aggregated API server].
+Even the https://github.com/crossplane/crossplane-runtime[crossplane-runtime] could be used to write the backend functionality, which is an SDK designed to especially work with external third-party APIs.
+
+A dedicated Kubernetes API server will be used for that to not interfere with other workload running on this instance.
+This could be a virtualized API server (for example https://github.com/loft-sh/vcluster[vcluster]) or a full-blown Kubernetes cluster.
+The controllers acting on the CRDs can run directly in this API server or outside in another Kubernetes cluster.
+All operations are handled through the Kubernetes API server, CRDs and controllers.
+
+image::decision/kubernetes-api-proposal.drawio.svg[]
+
+.An example architecture could look like this:
+[example]
+====
+* Third-party system connects directly to the metrics database and creates invoices.
+* Organizations are represented as `Organization` object in a dedicated APPUiO Cloud control namespace.
+  This then leads into the creation of a Group in Keycloak and a Namespace in the API server.
+* The Organization-level objects are stored in the Namespace of it and protected through appropriate RBAC rules:
+** Teams are represented as `Team` object and will manage the Sub-group in Keycloak
+** Invoices are represented as `Invoice` object, synced from the ERP system
+====
+
+== Decision
+
+Kubernetes API
+
+== Rationale
+
+The Kubernetes API approach has been chosen because the engineers at VSHN have a lot of know-how working with it.
+It allows to start fast as the important API plumbing is provided right from the beginning from the Kubernetes API server.
```

Comment:
```suggestion
It allows to start fast as the important API plumbing is provided right from the beginning from the Kubernetes API server.
And it adheres to the https://handbook.vshn.ch/hb/beliefs.html#_beliefs[VSHN Company Beliefs], mainly the part "For that we use".
```

Reply:
I removed this sentence because it's true in both cases. The REST API would also be running on Kubernetes...

Reply:
I think it's important to link to the Company Beliefs as this is an important reason for the decision.

Reply:
I've integrated this into the problem now, making it a precondition for any proposal.